opensquirrel/passes/router/shortest_path_router.py

```diff
@@ -0,0 +1,34 @@
+import networkx as nx
+
+from opensquirrel.ir import IR, SWAP, Gate
+from opensquirrel.passes.router import Router
+
+
+class ShortestPathRouter(Router):
+    def __init__(self, connectivity: dict[str, list[int]]) -> None:
+        self.connectivity = connectivity
+
+    def route(self, ir: IR) -> IR:
+        """
+        Routes the circuit by inserting SWAP gates to make it executable given the hardware connectivity.
+        Args:
+            ir: The intermediate representation of the circuit.
+        Returns:
+            The intermediate representation of the routed circuit (including the additional SWAP gates).
+        """
+        graph = nx.Graph({int(k): [int(v) for v in values] for k, values in self.connectivity.items()})
```

Comment:
The values should be integers already, right? So no need for a conversion. Also good to specify what the keys and values are.

```suggestion
        graph_data = {int(start_node): end_nodes for start_node, end_nodes in self.connectivity.items()}
        graph = nx.Graph(graph_data)
```